explainer.md

```diff
@@ -117,45 +119,20 @@ mediaSource.onsourceopen = (e) => {
   // App-specific request
   const mediaRequest = Uint8Array.from([1, 2, 3, 4]);
   const requestStream = await transport.createSendStream();
-  requestStream.write(mediaRequest);
+  const writer = requestStream.writable.getWriter();
+  await writer.ready();
+  writer.write(mediaRequest);
 
   transport.onreceivestream = (e) => {
-    const size = await readStreamUint32(e.stream);
-    if (!size || size > 512000) {
-      return;
-    }
-    const chunk = await readStreamBytes(e.stream, size);
-    if (!!chunk) {
-      sourceBuffer.appendBuffer(chunk);
+    const reader = e.stream.readable.getReader();
+    while (!reader.closed()) {
```

Comment:
`closed` is a promise, so instead I think instead you want something like:

```
while (true) {
  const {value: chunk, done} = await reader.read();
  if (done) break;
  sourceBuffer.appendBuffer(chunk);
}
```

Reply:
Done